### ISSUE TITLE
spontaneous `blockmsg` on p2p network doesn't get processed by chain/wallet

### ISSUE THREAD

**Issue by Christewart** [bitcoin-s/bitcoin-s#4894]
Observed on 5061fc6f

**Describe the bug**
When a block is mined on the network, a `BlockMessage` gets sent to us over the p2p network. It appears our chain/wallet do not process these correctly. This is occurring on mainnet. 

```
2022-11-21 20:16:13,202UTC INFO [PeerMessageSender] Requesting next compact filter headers from FilterSyncMarker(startHeight=764000, stopBlockHash=000000000000000000054548735df449582b4bbd9977d27809ec4492defd4a34)
2022-11-21 20:16:14,300UTC INFO [ChainHandler] Processed filters headers from height=764000 to 764177. Best filterheader.blockHash=000000000000000000054548735df449582b4bbd9977d27809ec4492defd4a34
2022-11-21 20:16:16,308UTC INFO [DataMessageHandler] Done syncing filter headers, beginning to sync filters from startHeightOpt=Some(764163)
2022-11-21 20:16:16,367UTC INFO [PeerMessageSender] Requesting compact filters from FilterSyncMarker(startHeight=764164, stopBlockHash=000000000000000000054548735df449582b4bbd9977d27809ec4492defd4a34)
2022-11-21 20:16:20,449UTC INFO [DataMessageHandler] Processing 14 filters
2022-11-21 20:16:20,536UTC INFO [ChainHandler] Processed filters from height=764164 to 764177. Best filter.blockHash=000000000000000000054548735df449582b4bbd9977d27809ec4492defd4a34
2022-11-21 20:16:20,539UTC INFO [DLCWallet$DLCWalletImpl] Processing 14 block filters for 0 utxos and 0 scripts
2022-11-21 20:16:20,554UTC INFO [DataMessageHandler] We are synced
2022-11-21 20:24:48,568UTC INFO [DataMessageHandler] Received headers message with 1 headers from Peer(neutrino.suredbits.com:8333)
2022-11-21 20:24:48,665UTC INFO [ChainHandler] Processed headers from height=764177 to 764178. Best hash=0000000000000000000650aa4cb6f4026c521e768cb086b09931616d21be1ee9
2022-11-21 20:24:48,671UTC INFO [DataMessageHandler] Starting to fetch filter headers in data message handler
2022-11-21 20:24:50,359UTC INFO [ChainHandler] Processed filters headers from height=764178 to 764178. Best filterheader.blockHash=0000000000000000000650aa4cb6f4026c521e768cb086b09931616d21be1ee9
2022-11-21 20:24:52,361UTC INFO [DataMessageHandler] Done syncing filter headers, beginning to sync filters from startHeightOpt=Some(764177)
2022-11-21 20:24:52,421UTC INFO [PeerMessageSender] Requesting compact filters from FilterSyncMarker(startHeight=764178, stopBlockHash=0000000000000000000650aa4cb6f4026c521e768cb086b09931616d21be1ee9)
2022-11-21 20:24:54,184UTC INFO [DataMessageHandler] Processing 1 filters
2022-11-21 20:24:54,197UTC INFO [ChainHandler] Processed filters from height=764178 to 764178. Best filter.blockHash=0000000000000000000650aa4cb6f4026c521e768cb086b09931616d21be1ee9
2022-11-21 20:24:54,203UTC INFO [DLCWallet$DLCWalletImpl] Processing 1 block filters for 0 utxos and 0 scripts
2022-11-21 20:24:54,235UTC INFO [DataMessageHandler] We are synced
//errors start here after IBD is done
2022-11-21 20:46:07,246UTC INFO [DataMessageHandler] Received block message with hash 0000000000000000000119faebfe2db08abed60a7176cc7126b097c78715e56e
2022-11-21 20:46:07,265UTC INFO [DLCWallet$DLCWalletImpl] Processing block=0000000000000000000119faebfe2db08abed60a7176cc7126b097c78715e56e heightOpt=None
2022-11-21 20:46:07,265UTC WARN [DLCWallet$DLCWalletImpl] Could not find blockheight for blockHash=0000000000000000000119faebfe2db08abed60a7176cc7126b097c78715e56e, skipping processing in wallet
2022-11-21 20:46:33,261UTC INFO [DataMessageHandler] Received block message with hash 000000000000000000073de23b525b5f46a7ffce002cea7776217ec8c5795219
2022-11-21 20:46:33,269UTC WARN [DLCWallet$DLCWalletImpl] Could not find blockheight for blockHash=000000000000000000073de23b525b5f46a7ffce002cea7776217ec8c5795219, skipping processing in wallet
2022-11-21 20:46:33,269UTC INFO [DLCWallet$DLCWalletImpl] Processing block=000000000000000000073de23b525b5f46a7ffce002cea7776217ec8c5795219 heightOpt=None
2022-11-21 20:56:53,837UTC INFO [DataMessageHandler] Received block message with hash 000000000000000000008406933d7b2ea9191b18d01955f79409c7579d352c85
2022-11-21 20:56:53,840UTC WARN [DLCWallet$DLCWalletImpl] Could not find blockheight for blockHash=000000000000000000008406933d7b2ea9191b18d01955f79409c7579d352c85, skipping processing in wallet
2022-11-21 20:56:53,840UTC INFO [DLCWallet$DLCWalletImpl] Processing block=000000000000000000008406933d7b2ea9191b18d01955f79409c7579d352c85 heightOpt=None
2022-11-21 21:13:19,712UTC INFO [DataMessageHandler] Received block message with hash 00000000000000000000eeb6250fce25e0e1ed9e62f879bf9837e805e4242299
2022-11-21 21:13:19,723UTC WARN [DLCWallet$DLCWalletImpl] Could not find blockheight for blockHash=00000000000000000000eeb6250fce25e0e1ed9e62f879bf9837e805e4242299, skipping processing in wallet
2022-11-21 21:13:19,723UTC INFO [DLCWallet$DLCWalletImpl] Processing block=00000000000000000000eeb6250fce25e0e1ed9e62f879bf9837e805e4242299 heightOpt=None
2022-11-21 21:16:48,766UTC INFO [DataMessageHandler] Received block message with hash 000000000000000000057418270228224c3aefc15192629a15590dcce58abdbe
2022-11-21 21:16:48,776UTC WARN [DLCWallet$DLCWalletImpl] Could not find blockheight for blockHash=000000000000000000057418270228224c3aefc15192629a15590dcce58abdbe, skipping processing in wallet
2022-11-21 21:16:48,776UTC INFO [DLCWallet$DLCWalletImpl] Processing block=000000000000000000057418270228224c3aefc15192629a15590dcce58abdbe heightOpt=None
2022-11-21 21:21:26,375UTC INFO [DataMessageHandler] Received block message with hash 00000000000000000005419b439da0381049bc3f49b22e516f91d6be4c4d4f7a
2022-11-21 21:21:26,379UTC WARN [DLCWallet$DLCWalletImpl] Could not find blockheight for blockHash=00000000000000000005419b439da0381049bc3f49b22e516f91d6be4c4d4f7a, skipping processing in wallet
2022-11-21 21:21:26,379UTC INFO [DLCWallet$DLCWalletImpl] Processing block=00000000000000000005419b439da0381049bc3f49b22e516f91d6be4c4d4f7a heightOpt=None
2022-11-21 21:25:00,688UTC INFO [BitcoinSServerMain$] @@@@@@@@@@@@@@@@@@@@@ Shutting down BitcoinSServerMain$ @@@@@@@@@@@@@@@@@@@@@
2022-11-21 21:25:00,689UTC ERROR [BitcoinSServerMain] Exiting process
2022-11-21 21:25:00,738UTC INFO [NodeCallbackStreamManager] Done draining akka streams for NodeCallbackStreamManager, it took=46ms
2022-11-21 21:25:00,746UTC INFO [HikariDataSource] bitcoin-s-node-db - Shutdown initiated...
2022-11-21 21:25:00,751UTC INFO [HikariDataSource] bitcoin-s-node-db - Shutdown completed.
2022-11-21 21:25:00,754UTC INFO [HikariDataSource] bitcoin-s.dlc.db - Shutdown initiated...
2022-11-21 21:25:00,754UTC INFO [P2PClientActor] We've been disconnected by Peer(neutrino.suredbits.com:8333) command=PeerClosed state=Normal
2022-11-21 21:25:00,755UTC INFO [HikariDataSource] bitcoin-s.dlc.db - Shutdown completed.
2022-11-21 21:25:00,759UTC INFO [HikariDataSource] bitcoin-s-chain-db - Shutdown initiated...
2022-11-21 21:25:00,760UTC INFO [HikariDataSource] bitcoin-s-chain-db - Shutdown completed.
2022-11-21 21:25:00,765UTC INFO [TorClient] No process found, binary wasn't started!
2022-11-21 21:25:00,772UTC INFO [WalletAppConfig] Stopping wallet rebroadcast task
2022-11-21 21:25:00,772UTC INFO [HikariDataSource] bitcoin-s-wallet-db - Shutdown initiated...
2022-11-21 21:25:00,773UTC INFO [HikariDataSource] bitcoin-s-wallet-db - Shutdown completed.

```

The wallet / chain seem to be stuck at block height `764,178` while bitcoin mainnet is at `764,184`.
```
bitcoin-s-cli getblockcount
764178
```

```
bitcoin-s-cli walletinfo
{                                                                                                                                                                                                           
  "wallet": {                                                                                         
    "keymanager": {                                                                                                                                                                                         
      "rootXpub": "zpub6jftahH18ngZwEj7UbnpNcQ5NjWBVK9p6c8kJVbiKkKvZLac2VmevzaBJUKUVckGUm4CvLgUag4fH63v2RV1UeoPhQyNomdSgx6Xi3E9Ptc"
    },                                                                                                                                                                                                      
    "walletName": "",
    "xpub": "zpub6rg7fS2WMhUsKeqvZwE5a7xmGHkHjDpFGgBm4d5QnTYYQDbU6JYF1GbGxsTMUFuB5VcuhD7c4CLSiVqUmG9pNcnusmUbpuKVR2UyvJWsxu9",                                                                              
    "hdPath": "m/84'/0'/0'",
    "height": 764178,                                                                                                                                                                                       
    "blockHash": "0000000000000000000650aa4cb6f4026c521e768cb086b09931616d21be1ee9",
    "rescan": false,
    "imported": false                                                                                                                                                                                       
  }
}
```

After restarting my node, sync seemed to occur fine. So it seems this is something wrong during steady state operations of the node.

**Comment by benthecarman** [comment]
Probably a bug with not processing the block header first to update it in the chain db

**Comment by Christewart** [comment]
Fixed by #5240
